treeherder/perf/management/commands/synthesize_backfill_report.py

```diff
@@ -0,0 +1,68 @@
+import datetime
+from argparse import ArgumentError
+
+from django.core.management.base import BaseCommand
+
+from treeherder.perf.alerts import (AlertsPicker,
+                                    IdentifyingRetriggerables)
+from treeherder.perf.models import PerformanceAlertSummary
+
+
+class Command(BaseCommand):
+    help = "Select most relevant alerts and identify jobs to retrigger."
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--time-window',
+            action='store',
+            type=int,
+            default=60
+        )
+
+        parser.add_argument(
+            '--frameworks',
+            nargs='+',
+            default=None
+        )
+
+        parser.add_argument(
+            '--repositories',
+            nargs='+',
+            default=None
+        )
+
+    def handle(self, *args, **options):
+        frameworks = options['frameworks']
+        repositories = options['repositories']
+        time_window = datetime.timedelta(minutes=options['time_window'])
+
+        max_timestamp = datetime.datetime.now() - time_window
+        one_day = datetime.timedelta(days=1)
+
+        summaries_to_retrigger = (PerformanceAlertSummary.objects
+                                  .select_related('framework', 'repository')
+                                  .filter(created__gte=max_timestamp))
+
+        if frameworks:
+            summaries_to_retrigger = summaries_to_retrigger.filter(framework__name__in=frameworks)
+            if len(set(frameworks) - set(summaries_to_retrigger)) > 0:
```

Comment:
This doesn't look correct. We should extract this validation into a helper method.

Reply:
All the code seems bloated at this point. I will break it into multiple methods.